### ISSUE TITLE
focus node 

### ISSUE THREAD

**Issue by mrqaidi** [lcuis/search_choices#76]
hi

how to connect to focus node from an textfield to dropdown

**Comment by lcuis** [comment]
Hello @mrqaidi ,

Thanks for this question.

I don't think there is a ready made solution to take focus on a `SearchChoices` Widget yet. Sorry about that.

I wonder whether wrapping it in a `FocusableActionDetector` Widget would be a potential solution?
Do you think this could work?

**Comment by mohamad-1994** [comment]
I have same problem,
in my app I have 1 text field and 2 search choice, I need autofocus to search choice when user click submit button on keyboard(in text filed) and then open second search choice after user select an item in first search choice.
I found setOpenDialog but didnt understand it clearly.

**Comment by lcuis** [comment]
Hello @mohamad-1994 ,

Let me rephrase your need.

You have the following interface:

- Text field
- SearchChoices1
- SearchChoices2

Here is the typical scenario:
1. Click text field
2. Enter value
3. Validate with return key
4. => populate items for SearchChoices1
5. => open SearchChoices1 with no item preselected nor any search keyword entered
6. Select item in SearchChoices1
7. => populate items for SearchChoices2
5. => open SearchChoices2 with no item preselected nor any search keyword entered

Is my understanding correct?

`setOpenDialog` seems to be the right way to go.

**Comment by mohamad-1994** [comment]
Hello @lcuis 
yes its correct, can you put an example for `setOpenDialog `? I didn`t understand it well.

**Comment by lcuis** [comment]
Thanks for the confirmation @mohamad-1994 , I can prepare you such an example when I will have access to a computer. In the meanwhile, you can have a look at the ’example/lib/main.dart’ which should contain an example with this argument.

**Comment by lcuis** [comment]
Hello @mohamad-1994 ,

I was able to run the following within the `example/lib/main.dart` file:

```dart
// ...
  Function? openFirstDialog,openSecondDialog;
// ...
      "Issue 76": Column(
        children: [
          TextField(
            onSubmitted: (String value){
              openFirstDialog!(value);
            },
          ),
          SearchChoices.single(
            items: items,
            value: selectedValueSingleDialog,
            hint: "Select one",
            searchHint: "Select one",
            onChanged: (value) {
              setState(() {
                selectedValueSingleDialog = value;
                openSecondDialog!(value);
              });
            },
            isExpanded: true,
            setOpenDialog: (openDialog){
              openFirstDialog=openDialog;
            },
          ),
          SearchChoices.single(
            items: items,
            value: selectedValueSingleDialog,
            hint: "Select one",
            searchHint: "Select one",
            onChanged: (value) {
              setState(() {
                selectedValueSingleDialog = value;
              });
            },
            isExpanded: true,
            setOpenDialog: (openDialog){
              openSecondDialog=openDialog;
            },
          ),
        ],
      ),
// ...
```

Here is the result I got:
![Screencast from 30-07-22 22_00_51 (2)](https://user-images.githubusercontent.com/32125299/181996268-2b14bfda-117d-46b4-903a-6aa0f64828f6.gif)

I hope this is close enough to what you need.

**Comment by lcuis** [comment]
Closing due to inactivity. Feel free to reopen if needed.
